client/src/components/Login/Login.js

```diff
@@ -0,0 +1,79 @@
+import React from 'react'
+import { Button, Box, TextField, Typography } from '@mui/material'
+import { useAuth } from '../Auth/AuthProvider'
+import { Navigate, useNavigate, useLocation } from 'react-router-dom'
+
+
+const Login = () => {
+  const { auth, login } = useAuth()
+  const navigate = useNavigate()
+  let location = useLocation()
+
+  // Assignment: redirect the newly logged in user to the page they were on
+  // OR to the User component
+
+  const source = location.state?.from?.pathname || "/User"
+  const backTologin = location.state?.from?.pathname || "/Login"
```

Comment:
What's the purpose of this line?

Reply:
I was thinking to create a conditional statement for the login but it was wrong fixed now.